core/src/main/java/org/apache/james/mime4j/codec/DecoderUtil.java

```diff
@@ -211,9 +232,20 @@ public static String decodeEncodedWords(String body, DecodeMonitor monitor, Char
     }
 
     // return null on error
-    private static String tryDecodeEncodedWord(final String mimeCharset,
-            final String encoding, final String encodedText, final DecodeMonitor monitor, final Charset fallback) {
+    private static String tryDecodeEncodedWord(
+            final String mimeCharset,
+            final String encoding,
+            final String encodedText,
+            final DecodeMonitor monitor,
+            final Charset fallback,
+            final Map<Charset, Charset> charsetOverrides) {
         Charset charset = CharsetUtil.lookup(mimeCharset);
+        if (charset != null && charsetOverrides != null) {
```

Comment:
could you please extract that code + the lookup into a well-named method?

Reply:
It's unclear to me what you mean by "that code". I pulled out all the charset related code into a method.